### PR TITLE
Asynchronous address book

### PR DESCRIPTION
**Pull request by lipis**
- https://github.com/wireapp/node-addressbook/pull/3
- https://github.com/wireapp/wire-desktop/pull/332

### CODE REVIEW

File: app/script/connect/ConnectRepository.coffee [hunk 1]
```diff
@@ -121,33 +120,33 @@ class z.connect.ConnectRepository
   @return [z.connect.PhoneBook] Encoded phone book data
   ###
   _parse_macos_contacts: ->
-    return if not window.zAddressBook
-
-    address_book = window.zAddressBook()
-    phone_book = new z.connect.PhoneBook @properties_repository.self()
+    return new Promise (resolve, reject) =>
+      return resolve undefined if not window.zAddressBook
+      address_book = window.zAddressBook
```
Comment (herrmannplatz) [inline app/script/connect/ConnectRepository.coffee]: keep in mind that this breaks older clients

Comment (lipis) [reply app/script/connect/ConnectRepository.coffee]: right.. shall we rename the `zAddressBook` to `wAddressBook` maybe? That should work.. and also maybe removing the option to import from AddressBook if that module is missing..

Comment (gregor) [reply app/script/connect/ConnectRepository.coffee]: Do we want to shim the old one in webapp for a bit of time maybe? Version update on macOS takes a while.

Comment (herrmannplatz) [reply app/script/connect/ConnectRepository.coffee]: @lipis old mac clients will still expose zAddressbook as a Function, so you should take that into account by checking for the new and the old syntax.

**Pull request merged**